ubcpi/static/js/spec/ubcpi_edit_spec.js

```diff
@@ -381,7 +381,76 @@ describe('UBCPI_Edit module', function () {
                 expect(mockNotify.calls.count()).toBe(3);
             });
         });
-    })
+    });
+    describe('EditSettingsController', function () {
+        var $rootScope, createController, controller, expectedCorrectRationale;
+
+        beforeEach(inject(function ($controller, _$rootScope_) {
+            mockConfig.data = {
+                "image_position_locations": {"below": "Appears below", "above": "Appears above"},
+                "rationale_size": {"max": 32000, "min": 1},
+                "display_name": "Peer Instruction",
+                "algo": {"num_responses": "#", "name": "simple"},
+                "algos": {
+                    "simple": "System will select one of each option to present to the students.",
+                    "random": "Completely random selection from the response pool."
+                },
+                "correct_answer": 1,
+                "seeds": [
+                    {answer:2, rationale:'rationale3'},
+                    {answer:1, rationale:'rationale2'},
+                    {answer:0, rationale:'rationale1'},
+                    {answer:2, rationale:'rationale3'},
+                    {answer:1, rationale:'rationale2'},
+                    {answer:0, rationale:'rationale1'}
+
+                ],
+                "question_text": {
+                    "text": "What is the answer to life, the universe and everything?",
+                    "image_show_fields": 0,
+                    "image_url": "",
+                    "image_position": "below",
+                    "image_alt": ""
+                },
+                "options": [
+                    {
+                        "text": "21",
+                        "image_show_fields": 0,
+                        "image_url": "",
+                        "image_position": "below",
+                        "image_alt": ""
+                    }
+                ]
+            };
+
+            $rootScope = _$rootScope_;
+            createController = function (params) {
+                return $controller(
+                    'EditSettingsController', {
+                        $scope: $rootScope,
+                        $stateParams: params || {}
+                    });
+            }
+        }));
+
+        beforeEach(function() {
```

Comment:
Why are there two `beforeEach` calls?

Reply:
This can be done in a single `beforeEach` definition but for better readability and staying consistent with the existing similar test_suite, there are two definitions. The definitions are called from top to bottom.

Reply:
They should be moved to one. This doesn't affect readability but adds a sense of ambiguity as a function has been defined twice with different content.